### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by sahlminkok**
### In this Pull Request I:
- Add Redux Toolkit
- Create a directory that will contain all the Redux logic
- Configure a Redux store
- Define a slice of state for books
- Define a slice of state for categories

### CODE REVIEW

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { configureStore } from '@reduxjs/toolkit';
+import booksReducer from './books/booksSlice';
+
+const store = configureStore({
+  reducer: {
+    book: booksReducer,
+  },
+});
+
+export default store;
```
Comment (Ibizugbe) [inline src/redux/store.js]: Nice job adding the bookReducer to your store👍. currently, you haven't added the categoriesReducer😉. Kindly add it as this is a requirement in this milestone👍.

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      const newBook = {
+        id: 4,
+        title: 'Book',
+        category: 'cat',
+        author: 'me',
+        completed: 45,
+        chapter: 'chapter 89',
+      };
+      state.booksList = [...state.booksList, newBook];
+    },
```
Comment (Ibizugbe) [inline src/redux/books/booksSlice.js:21]: Nice job so far👍. Just as required for this milestone, the addBook reducer should currently return an empty array😉.
image
I suggest you remove the books you have added and configure the addBook reducer to return an empty array of books as required👍.

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @sahlminkok 👋👋👋, 
tryagain


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are correctly setup and they are passing👍.
- [x] Giflow is used👍.
- [x] commit messages are descriptive and meaningful👍.
- [x] File structure follows the duck pattern👍.
- [x] @reduxjs/toolkit & react-redux are present in package.json👍.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A😊

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 2 above)
Comment (sahlminkok) [reply src/redux/books/booksSlice.js:21]: @Ibizugbe but the initial state is empty I guess and here I'm just declaring a new state after dispatch, please consider and guide me through how I'm going to make this work.

**Review by divinecharlotte — APPROVED** [review]
# Hello @sahlminkok     😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say  👍🏻 ✅ 
### To Highlight 💯


- Descriptive PR👍 
- Good implementation of the required changes 👍 
 




## Congratulations! 🎉 



Cheers, and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**